### DISCUSSION TITLE
Advise requested regarding restore

### DISCUSSION THREAD

**Discussion by Alextocode** [stashapp/stash#5569] [Q&A]
Hi there,

I am running Stashapp on an Ubuntu VM. When I upgraded that VM from Ubuntu 22 to 24 something seems to have gone seriously wrong and it spoiled the whole VM. To make things worse I found that the backup of this VM I had was spoil tool, probably due to some network error when backing it up last time.

However I have the virtual disk file of the old server and it seems data look okay. So I got the drive including blobs, data, config, metadata directory, etc, with the db in sqlite format and the config.yml - basically everything - I can just not start the server using this disk anymore, but I can access the data if I create a new Ubuntu server and mount it as drive ( I mapped to /mnt/stashold). And I found that config.yml has paths set that are not used apparently like /root/.stash/stash-go.sqlite as database path, while the database is actually inside the stashapp folder in config directory but it was like this on the old server as well and worked so probably not an issue.

I tried now to just setup a fresh Stash instance and restore everything as it was before, but I am unsure how to do it correctly (you may beat me for that stupid question, because actually I am a computer scientist myself - in my defence, I am working only with Windows servers in my job and try to avoid Linux usually.

Does anyone have experience on how to make that working again? TBH my preferred scenario would **actually be** to have two virtual disk drives = two VHDX files - one for the OS, one for Stashapp data only). But I am not sure how the syntax would have to look like to point to a location that is not inside the stashapp folder (but to the mounted drive).

So in other words - and for the perfect solution for me - my question is:

How can I setup a new Stash instance but use everything from the mounted drive 1:1 - so basically when I start Stash again nothing is lost? Probably if you tell me the format for the config.yml it might already do the trick. I want to avoid that Stash has to generate everything again, since I already got more than 400GB of video content.

Any hints or guidance is warmly welcome. I am too worried to make something wrong, so I dare to just ask the community. Probably there are some Linux geeks out there that will be amused - but probably also be able to help :)

Cheers!

Alex

**Comment by feederbox826** [comment]
Here's the migration script for my shell script, if you want to play it as safe as possible, create a new stash on your os/data pair, set up the basic paths and you should be able to copy and compare with your old paths

https://github.com/feederbox826/stash-s6/blob/main/stash/root/opt/entrypoint.sh#L192-L239

it should be simple as, if your stashapp instance is on /mnt/stash-data, then replacing `/root/.stash` with `/mnt/stash-data`

**Reply by Alextocode** [reply]
Thanks so much, @feederbox826 , this seems to be a quite complex script. Would you mind to talk a word or two about what it does in detail? I understand it copies everything from the /mnt/stash-data path back to home/{username}/stashapp/ correct? Does this also affect the database etc?
Or in other words, what was the purpose this script was written for? Apologize my curiosity but I would like to make sure the script does what I need it to do :)

Thanks again for your input already given, appreciate it.

**Comment by feederbox826** [comment]
> Or in other words, what was the purpose this script was written for?
basically automated migration

the migration steps are (generally) as follows:

1. check old path valid
2. check new path valid
3. move files from old path to new path
4. update value in config.yml to reflect new path instead of old path
5. move files that don't need to me changed in config.yml

check_migrate takes the following arguments config.yml key, old path, new path, fallback

if you want to manually migrate, just move the files manually and check config.yml for the corresponding key name and reflect it. the remaining files under misc files don't need this config.yml check
